src/Shared/BuildEnvironmentHelper.cs

```diff
@@ -599,7 +599,7 @@ public BuildEnvironment(BuildEnvironmentMode mode, string currentMSBuildExePath,
 
                 MSBuildToolsDirectory32 = MSBuildToolsDirectoryRoot;
                 MSBuildToolsDirectory64 = existsCheck(potentialAmd64FromX86) ? Path.Combine(MSBuildToolsDirectoryRoot, "amd64") : CurrentMSBuildToolsDirectory;
-                MSBuildToolsDirectoryArm64 = existsCheck(potentialARM64FromX86) ? Path.Combine(MSBuildToolsDirectoryRoot, "arm64") : null;
+                MSBuildToolsDirectoryArm64 = existsCheck(potentialARM64FromX86) ? Path.Combine(MSBuildToolsDirectoryRoot, "arm64") : CurrentMSBuildToolsDirectory;
```

Comment:
I think this change should apply only to .NET Core MSBuild, to prevent a problem where:

1. On an amd64 VS installation (no arm64 folder)
2. UsingTask specifies that a task should run in arm64

With this, I'm worried that we will silently run the task in amd64 instead of correctly failing with a "we don't have arm64" error.

Reply:
This wouldn't apply the same way to x86, because that's "always" available in a VS installation.